### PR TITLE
Access a literal wide string after a memory growth [embind]

### PR DESCRIPTION
**Pull request by canning-duck**
Fix the issue #3299.

*Important*: I've added "-s ALLOW_MEMORY_GROWTH=1" to the compilation of the tests in test_embind() of tests/test_other.py.

Tests : other.test_embind + the following random 100:

* asm1i.test_strstr
* asm2g.test_emscripten_get_compiler_setting
* asm3i.test_stack_varargs
* asm2.test_stdvec
* asm3i.test_trickystring
* asm2nn.test_longjmp_throw
* asm2nn.test_ptrtoint
* asm2nn.test_dynamic_cast_2
* asm3i.test_lua
* asm2nn.test_fuzz
* default.test_atomic
* asm2g.test_unsigned
* asm2.test_locale
* asm3i.test_bigarray
* default.test_cases
* asm2.test_sintvars
* asm2f.test_parseInt
* asm3.test_safe_dyncalls
* asm3i.test_stack_align
* asm2g.test_memmove3
* asm3.test_inlinejs3
* asm3i.test_memcpy3
* asm2g.test_exceptions_refcount
* asm1.test_systypes
* asm2f.test_dlfcn_mallocs
* asm1.test_openjpeg
* asm1.test_longjmp_funcptr
* asm3.test_unicode_js_library
* asm3.test_cases
* asm2.test_strcmp_uni
* asm2f.test_multiply_defined_symbols
* asm1i.test_corruption_3
* asm2.test_exceptions_std
* asm2.test_simd5
* asm1i.test_alloca
* asm2g.test_tracing
* asm1i.test_std_cout_new
* asm3.test_frexp
* asm2.test_sintvars
* asm2.test_the_bullet
* asm2nn.test_unistd_sleep
* asm2f.test_systypes
* asm2g.test_sizeof
* asm3i.test_funcs
* asm2.test_sscanf_3
* asm2nn.test_strstr
* asm2nn.test_funcptr_namecollide
* asm1i.test_systypes
* asm2nn.test_i64_umul
* asm3i.test_double_varargs
* asm1i.test_math_lgamma
* asm2nn.test_inherit
* asm3.test_unsigned
* asm1i.test_atexit
* asm1i.test_stat
* asm3.test_line_endings
* asm2g.test_getopt
* asm1i.test_dlfcn_data_and_fptr
* asm2.test_sscanf_skip
* asm2nn.test_fs_emptyPath
* asm2f.test_stack_overflow
* asm1i.test_errar
* asm3.test_sscanf_whitespace
* asm2nn.test_tcgetattr
* asm3i.test_alloca
* default.test_freetype
* asm1.test_simd5
* asm2f.test_environ
* asm3i.test_unicode_js_library
* asm2g.test_zerodiv
* default.test_inherit
* asm1.test_polymorph
* asm1.test_time
* asm3.test_istream
* asm2.test_math
* asm2f.test_corruption
* asm1i.test_strptime_reentrant
* asm2nn.test_random
* asm2g.test_openjpeg
* asm1i.test_fasta
* asm3.test_memcpy3
* asm1.test_unistd_misc
* asm1i.test_unistd_login
* asm3i.test_strtod
* asm2f.test_strtol_bin
* asm3.test_trickystring
* asm2nn.test_dlfcn_varargs
* default.test_flexarray_struct
* asm2f.test_locale
* asm2nn.test_embind_2
* asm3.test_fast_math
* asm3.test_the_bullet
* asm3.test_dlfcn_qsort
* asm1.test_nestedstructs
* asm3.test_unistd_close
* asm3.test_strndup
* asm2nn.test_fcvt
* asm1i.test_trickystring
* default.test_strftime
* asm2g.test_dynamic_cast
* asm1.test_complex

### CODE REVIEW

File: tests/test_other.py [hunk 1]
```diff
@@ -2185,6 +2185,7 @@ def test_embind(self):
           path_from_root('tests', 'embind', 'embind_test.cpp'),
           '--pre-js', path_from_root('tests', 'embind', 'test.pre.js'),
           '--post-js', path_from_root('tests', 'embind', 'test.post.js'),
+          '-s', 'ALLOW_MEMORY_GROWTH=1',
```
Comment (kripken) [inline tests/test_other.py]: let's test with and without memory growth (add it like closure is added above)

**Comment by canning-duck** [comment]
force_memory_growth() raises an "abort" exception and makes the test fail if memory growth is deactivated. is there a way to check this is the case / avoid the failure? try/catch catches the exception but does not prevent the test failure.

**Comment by kripken** [comment]
Hmm, there isn't really a good way to know if memory growth is enabled. Except that when you try to grow memory, it will throw with a message saying you should enable memory growth. Could a try-catch check if it sees that message, and if so assume memory growth is off, and ignore that part of the test?

**Comment by canning-duck** [comment]
A try-catch catches the exception but the testing framework decides the test is a failure anyway. Maybe the framework has an exception handler registered somewhere. I'll look further into that.

**Comment by canning-duck** [comment]
@kripken Here's a better version of the tests (hopefully). The try-catch was not the issue, the failed memory growth was (another test was failing later because of it).

**Comment by kripken** [comment]
Tests look good, thanks. I assume you verified they pass?

If @chadaustin is ok with the embind fix, we can merge this.

**Comment by chadaustin** [comment]
Looks good to me!  (Sorry, I meant to comment on this a long time ago!)

**Comment by kripken** [comment]
Merged, thanks!

**Pull request merged**